pkg/ocm/ocm.go

```diff
@@ -237,9 +239,57 @@ func (c *SdkClient) LimitedSupportExists(ls LimitedSupportReason, internalCluste
 	return false, nil
 }
 
+// isLimitedSupportReasonFlapping checks if a cluster has been put in limited support too many times
+// Returns true if the cluster has been put in limited support at least 2 times in the last 4 hours
+func (c *SdkClient) isLimitedSupportReasonFlapping(ls LimitedSupportReason, internalClusterID string) (bool, error) {
+	// Retrieve external ID
+	info, err := c.GetClusterInfo(internalClusterID)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve cluster info: %w", err)
+	}
+
+	externalID := info.ExternalID()
+
+	// Rerieve service logs
+	resp, err := c.conn.ServiceLogs().V1().Clusters().Cluster(externalID).ClusterLogs().List().Send()
+	if err != nil {
+		return false, fmt.Errorf("could not list service logs history: %w", err)
+	}
+
+	serviceLogs, ok := resp.GetItems()
+	if !ok {
+		return false, nil
+	}
+
+	// Check if the cluster has been put in limited support at least 2 times in the last 4 hours
+	fourHoursAgo := time.Now().Add(-4 * time.Hour)
+
+	var filteredLogs []*servicelogsv1.LogEntry
+	for _, log := range serviceLogs.Slice() {
+		if log.ServiceName() == "LimitedSupport" &&
+			log.Username() == "service-account-ocm-cad-production" &&
+			log.Timestamp().After(fourHoursAgo) &&
+			log.Summary() == ls.Summary {
+			filteredLogs = append(filteredLogs, log)
+		}
+	}
+
+	return len(filteredLogs) >= 2, nil
+}
+
 // DeleteLimitedSupportReasons removes *all* limited support reasons for a cluster which match the given summary
 // Returns true if a limited support reason got removed
 func (c *SdkClient) DeleteLimitedSupportReasons(ls LimitedSupportReason, internalClusterID string) (bool, error) {
+	isFlapping, err := c.isLimitedSupportReasonFlapping(ls, internalClusterID)
+	if err != nil {
+		return false, err
+	}
+
+	if isFlapping {
+		logging.Infof("Limited support reason is flapping `%s`", ls.Summary)
+		return false, nil
+	}
+
```

Comment:
**issue:** `DeleteLimitedSupportReasons` currently does not document any of this behavior. 

**suggestion:** the flapping timespan could be a parameter of `DeleteLimitedSupportReasons`, and should be documented in the functions comment. Value `0` for the flappyness span could be "always delete the reason".

Reply:
Can you elaborate more on why we need it as a parameter and maybe give an example of a case in which you want to add and remove limited support multiple times?

Reply:
Currently, a customer can trigger CHGM by stopping nodes. The main reason we need this change is that a the nodes stopped by the customer usually get restarted by hive. This means that if the customer has automation in place to stop nodes (e.g. some AWS lambda), we have hive fighting the lambda, and the customer is spammed by limited support emails.

For other alerts/investigations (in the future), we should not have that problem, and we could default to putting clusters in and out of limited support purely based on the actions of the customer. 

Adding the parameter also makes it clearer to the caller that this is happening "under the hood". 

If you don't want to add this as parameter, I'm also okay with just documenting the behavior clearly in `DeleteLimitedSupportReasons`.

Reply:
I documented the behavior, I think the parameter can be easily added when it is needed.